### PR TITLE
#163017657 authors haven database models

### PR DESCRIPTION
**Pull request by Proception**
#### What does this PR do?
It creates all the database models using sequelize and adds all migrations.
#### Description of Task to be completed?
- define database models in sequelize
- create migration files for models
- specify relationships between models

#### How should this be manually tested?
- run `npm install`
- run `npm run db:unmigrate`
- run `npm run db:migrate`
- appropraite tables shoud be created

#### Any background context you want to provide?
Check database tables to see contraints specified on the models and relationships

#### What are the relevant pivotal tracker stories?
[#163017657](https://www.pivotaltracker.com/story/show/163017657)

#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/db/migrations/20190114155001-create-art.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Arts', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      userId: {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190114155001-create-art.js]: Can we be more descriptive about this? Something like `artistId` should work.

File: src/db/migrations/20190114155001-create-art.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      },
+      categoryId: {
+        type: Sequelize.INTEGER,
+        references: {
+          model: 'Categories',
+          key: 'id',
+          as: 'categoryId',
+        }
+      },
+      featuredImg: {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190114155001-create-art.js:35]: How do you intend to save the rest of the images?

File: src/db/migrations/20190114160429-create-media.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      contentUrl: {
+        type: Sequelize.TEXT
+      },
+      mediaType: {
+        type: Sequelize.STRING
+      },
+      artId: {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190114160429-create-media.js:16]: Why is there a relationship between an Art and a media?

File: src/db/migrations/20190115144953-create-follow-summary.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190115144953-create-follow-summary.js:1]: Why do you need  this model?

File: src/db/migrations/20190115145052-create-like-summary.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+module.exports = {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190115145052-create-like-summary.js:1]: Why do you need this as well?

File: src/db/migrations/20190115160348-create-updated-comment.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+module.exports = {
```
Comment (andela-dbamidele) [inline src/db/migrations/20190115160348-create-updated-comment.js:1]: What's this for?

File: src/db/models/art.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    Art.hasMany(models.Comment, {
+      foreignKey: 'artId'
+    });
+    Art.hasMany(models.Like, {
+      foreignKey: 'artId'
+    });
+    Art.hasOne(models.LikeSummary, {
+      foreignKey: 'artId'
+    });
+    Art.hasOne(models.Transaction, {
```
Comment (andela-dbamidele) [inline src/db/models/art.js:32]: Why  `hasOne`?

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/db/migrations/20190114155001-create-art.js (hunk 2 above)
Comment (Proception) [reply src/db/migrations/20190114155001-create-art.js:35]: The rest of the images will be saved on the media table

File: src/db/migrations/20190114160429-create-media.js (hunk 3 above)
Comment (Proception) [reply src/db/migrations/20190114160429-create-media.js:16]: Art has a maximum of 6 images and one video. To abstract it, we created a media table to store all media(images and video) for the Art model. That way for an Art item we can pull all records from the media table.

File: src/db/migrations/20190115144953-create-follow-summary.js (hunk 4 above)
Comment (Proception) [reply src/db/migrations/20190115144953-create-follow-summary.js:1]: We need a way to store the summary data for each user. This record will hold the calculated value of the number of followers and the number of people following the user. This saves us the performance overhead of having to calculate these values every time the data is needed.

File: src/db/migrations/20190115145052-create-like-summary.js (hunk 5 above)
Comment (Proception) [reply src/db/migrations/20190115145052-create-like-summary.js:1]: We need a way to store the summary data for each user. This record will hold the calculated value of the number of likes for an art. This saves us the performance overhead of having to calculate these values every time the data is needed.

File: src/db/migrations/20190115160348-create-updated-comment.js (hunk 6 above)
Comment (Proception) [reply src/db/migrations/20190115160348-create-updated-comment.js:1]: There is a task that specifies that we need to keep track of comment edit history. To do this we needed another table called updated comment. Therefore all changes are recorded on this table.

File: src/db/models/art.js (hunk 7 above)
Comment (Proception) [reply src/db/models/art.js:32]: Art has only one like summary. The like summary will hold a summary of likes for an article

File: src/db/migrations/20190114155001-create-art.js (hunk 1 above)
Comment (Proception) [reply src/db/migrations/20190114155001-create-art.js]: Yes, that is better. Will implement and revert.

**Pull request merged**